lib/AST/ASTContext.cpp

```diff
@@ -670,8 +670,7 @@ FuncDecl *ASTContext::getPlusFunctionOnRangeReplaceableCollection() const {
         continue;
       for (auto Req: FD->getGenericRequirements()) {
         if (Req.getKind() == RequirementKind::Conformance &&
-              Req.getSecondType()->getNominalOrBoundGenericNominal() ==
-            getRangeReplaceableCollectionDecl()) {
+              Req.getSecondType()->isRangeReplaceableCollection()) {
```

Comment:
This has me thinking that the "Type" suffix might be valuable, to not imply that this is doing a compatibility check. `Array<Int>` "is a RangeReplaceableCollection" in some sense.

Reply:
There are a couple of protocols listed in KnownStdlibTypes that could be moved over to KnownProtocols, that way they would still have to compare against the decl. We could introduce a isProtocolType for those.